Sources/ManageMembergroups.php

```diff
@@ -771,6 +797,14 @@ function EditMembergroup()
 			foreach ($accesses as $group_id => $action)
 				$changed_boards[$action][] = (int) $group_id;
 
+			$smcFunc['db_query']('', '
+				DELETE FROM {db_prefix}board_permissions_view
+				WHERE id_group = {int:group_id}',
+				array(
+					'group_id' => (int) $_REQUEST['group'],
+				)
+			);
```

Comment:
For the sake of security, it would be better if the DELETE query had a condition to make sure it only acted on boards that we know for certain are going to have new values inserted for them.

Reply:
see the msg below,
we don't know which boards got deleted,
we only know which got added.
And for the "added", we don't know if they are changed or not.

For make sure that the data 100% sync with the old format data,
is the way right here.

In the worst case had a group no access to boards,
but keep able to access admin side to fix things --> no special critcal checks needed.

Reply:
I suppose you are right.

My concern was that if we deleted all entries for a group and then the submitted data only set permissions for some boards for that group, it would be possible to clear the deny permissions so that a group that was formerly blocked from a certain board could potentially gain access to it again.

But since we can rely on the security checks to prevent random hackers from making these changes, I suppose I was just being paranoid. So never mind. 🙂